lib/matplotlib/dates.pyi

```diff
@@ -0,0 +1,200 @@
+import datetime
+import numpy as np
+from typing import Any, Callable, Dict, List, Sequence
+
+from .axis import Axis
+from .projections.polar import _AxisWrapper
+from .ticker import _DummyAxis, Formatter, Locator
+from .units import AxisInfo, ConversionInterface
+
+
+# --- Global Functions ---
+
+def _get_tzinfo(tz: str | datetime.tzinfo | None = ...) -> datetime.tzinfo: ...
+
+def _reset_epoch_test_example() -> None: ...
+
+def set_epoch(epoch: str) -> None: ...
+
+def get_epoch() -> str: ...
+
+def _dt64_to_ordinalf(d: np.datetime64 | np.ndarray) -> float | np.ndarray: ...
+
+def _from_ordinalf(x: float, tz: str | datetime.tzinfo | None = ...) -> datetime.datetime: ...
+
+def datestr2num(d: str | Sequence[str], default: datetime.datetime | None = ...) -> float | np.ndarray: ...
+
+def date2num(d: datetime.datetime | np.datetime64 | Sequence[datetime.datetime | np.datetime64]) -> float | np.ndarray: ...
+
+def num2date(x: float | Sequence[float], tz: str | datetime.tzinfo | None = ...) -> datetime.datetime | list[datetime.datetime]: ...
+
+def num2timedelta(x: float | Sequence[float]) -> datetime.timedelta | list[datetime.timedelta]: ...
+
+def drange(dstart: datetime.datetime, dend: datetime.datetime, delta: datetime.timedelta) -> np.ndarray: ...
+
+def _wrap_in_tex(text: str) -> str: ...
+
+# --- Formatter Classes ---
+
+class DateFormatter(Formatter):
+    tz: str | datetime.tzinfo | None
```

Comment:
This is passed through `_get_tz_info` before assignment, so should always be a `datetime.tzinfo`.